### ISSUE TITLE
Switch JSON to RapidJSON (bot)

### ISSUE THREAD

**Issue by viniciusdc** [conda-forge/conda-forge-bot#1112]
So, as an attempt to do this replacement I was wondering to start with `auto_tick.py` as a reference.

The only explicit usage of JSON there is on `initialize_migrations` function, more especifilacy at 

```python 
pinning_version = json.loads(eval_cmd("conda list conda-forge-pinning --json"))[0]["version"]
```
And of course we have the issue relating the LaizyJSON class, which IDK if it's necessary to change.

**Comment by CJ-Wright** [comment]
I wouldn't worry about the pinning_version, the json that you need to move is inside `utils`.

**Comment by viniciusdc** [comment]
> I wouldn't worry about the pinning_version, the json that you need to move is inside `utils`.

Thank you very much CJ

**Comment by CJ-Wright** [comment]
yeah see if that produces the needed results and we can shop around

**Comment by viniciusdc** [comment]
> yeah see if that produces the needed results and we can shop around

I know that explicity is better (in this case even more) but what if I use `import rapidjson as json` will it works ?

**Comment by viniciusdc** [comment]
Ok, I tested it with `test_utils` was it enought as a code behavior test ?

**Comment by CJ-Wright** [comment]
I would try to load and dump the entire graph

**Comment by viniciusdc** [comment]
ok

**Comment by viniciusdc** [comment]
`load_graph` worked fine, but for `dump_graph` I received an error message:
```python
File "E:\PycharmProjects\ts-graph\conda_forge_tick\utils.py", line 462, in dump_graph
    dump_graph_json(gx, filename)
  File "E:\PycharmProjects\ts-graph\conda_forge_tick\utils.py", line 435, in dump_graph_json
    dump(nld, f)
  File "E:\PycharmProjects\ts-graph\conda_forge_tick\utils.py", line 404, in dump
    return json.dump(
TypeError: 'separators' is an invalid keyword argument for rapidjson.dump()
```

**Comment by viniciusdc** [comment]
By the reference of rapidJSON at PyPi ([see](https://pypi.org/project/python-rapidjson/) for instance) it looks like we don't have the separators support.

**Comment by viniciusdc** [comment]
which in our case is not a good news right? Almost all the nodes attrs has it on their names

**Comment by CJ-Wright** [comment]
Well... that gives us a few options:
1. We could choose a different json loader
2. We could modify the existing graph to not use seperators

**Comment by viniciusdc** [comment]
> Well... that gives us a few options:
> 
> 1. We could choose a different json loader
> 2. We could modify the existing graph to not use seperators

Even though I know the first option will be interesting (but I think that, if that it the case we should just let as it is by now).  I like the second one more, cause it will be possible to rethink some other ideas too, also sometimes I have errors with `load_graph` because these separators (for instance `conda_forge_auto_tick::git.json`)

**Comment by CJ-Wright** [comment]
Ok, in that case please write up as script in your PR to reformat the database as needed. Before the code goes in we'll halt the bot and run the script.

**Comment by viniciusdc** [comment]
> Ok, in that case please write up as script in your PR to reformat the database as needed. Before the code goes in we'll halt the bot and run the script.

Ok, I will try. In this PR ?

**Comment by CJ-Wright** [comment]
This isn't a PR, but yes that should be in the same PR as the format change

**Comment by viniciusdc** [comment]
> This isn't a PR, but yes that should be in the same PR as the format change

Sorry I forgot that this was an issue, rssr. Ok, I will work on that

**Comment by viniciusdc** [comment]
Okay, some news... I've thinking during the week about these changes and will update the here soon. It fundamentally consists of three big changes:
- 1. The decorators, as we need to replace that to enable `RapidJSON` usage;
- 2. The `run` function, as a matter of optimizations, I think that looking for modularity on this subject is a good starting point;
- 3. The last but more difficult, to create an independent subprocess for PR management and tracking. Why? Well, this way we can lessen the stress to the bot (PR/migrations) and pinning, as we will be able to handle more precisely the error messages to insert more context and directions to the maintainers.

**Comment by mariusvniekerk** [comment]
there are some python mappings to simdjson which we could give a try

**Comment by viniciusdc** [comment]
Thanks marius, I will have a look. Also I still need to test the bump methods without the decorators... to have an ideia if it will work.

**Comment by beckermr** [comment]
We use orjson now. Closing this.
